### Data Analysis Project -- Indian Start-up Funding Analysis

### BUSINESS UNDERSTANDING

### Ideas, creativity, and execution are essential for a start-up to flourish. But are they enough? Investors provide start-ups and other entrepreneurial ventures with the capital---popularly known as "funding"---to think big, grow rich, and leave a lasting impact. 

### In this project, you are going to analyse funding received by start-ups in India from 2018 to 2021. You will find the data for each year of funding in a separate csv file in the dataset provided. 

### In these files you'll find the start-ups' details, the funding amounts received, and the investors' information.



### Scenario

### Your team is trying to venture into the Indian start-up ecosystem. As the data expert of the team you are to investigate the ecosystem and propose the best course of action.

### Instructions

### Your task is to develop a unique story from this dataset by stating and testing a hypothesis, 

### asking questions, perform analysis and share insights with appropriate visualisations.

### So as part of the project you are to:

### 1) Ask questions

### 2) Develop hypothesis

### 3) Process the data

### 4) Analyse the data

### 5) Visualise the data



### UNDERSTANDING YOUR DATA

### INPORTING LIBRARIES

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

import plotly.express as px
import plotly.graph_objs as go


### LOADING DATASETS

In [2]:
startup_2018= pd.read_csv(r'C:\Users\User\Desktop\AZUBI AFRICA\Projects\LP 1\startup_funding2018.csv')

startup_2018
                           


Company Name                                           Industry  \
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

    Round/Series       Amount                          Location  \
0           Seed       250000       Bangalore, Karnataka, India   
1           Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2       Series A  ₹65,000,000           Gurgaon, Haryana, India   
3          Angel      2000000       Noida, Uttar Pradesh, India   
4           Seed            —  Hyderabad, Andhra Pradesh, India   
..           ...          ...                               ...   
521     Series C    225000000       Bangalore, Karnataka, India   
522     Series A            —           Haryana, Haryana, India   
523         Seed         7500        Mumbai, Maharashtra, India   
524         Seed  ₹35,000,000        Mumbai, Maharashtra, India   
525     Series C     35000000        Chennai, Tamil Nadu, India   

                                         About Company  
0    TheCollegeFever is a hub for fun, fiesta and f...  
1    A startup which aggregates milk from dairy far...  
2            Leading Online Loans Marketplace in India  
3    PayMe India is an innovative FinTech organizat...  
4    Eunimart is a one stop solution for merchants ...  
..                                                 ...  
521  Udaan is a B2B trade platform, designed specif...  
522            HappyEasyGo is an online travel domain.  
523  Mombay is a unique opportunity for housewives ...  
524  Droni Tech manufacture UAVs and develop softwa...  
525       Welcome to India's most convenient pharmacy!  

[526 rows x 6 columns]

In [3]:
startup_2019 = pd.read_csv(r'C:\Users\User\Desktop\AZUBI AFRICA\Projects\LP 1\startup_funding2019.csv')

startup_2020 = pd.read_csv(r'C:\Users\User\Desktop\AZUBI AFRICA\Projects\LP 1\startup_funding2020.csv')

startup_2021 = pd.read_csv(r'C:\Users\User\Desktop\AZUBI AFRICA\Projects\LP 1\startup_funding2021.csv')

startup_2019

Company/Brand  Founded HeadQuarter                   Sector  \
0    Bombay Shaving      NaN         NaN                Ecommerce   
1         Ruangguru   2014.0      Mumbai                   Edtech   
2          Eduisfun      NaN      Mumbai                   Edtech   
3          HomeLane   2014.0     Chennai          Interior design   
4          Nu Genes   2004.0   Telangana                 AgriTech   
..              ...      ...         ...                      ...   
84     Infra.Market      NaN      Mumbai                Infratech   
85              Oyo   2013.0    Gurugram              Hospitality   
86       GoMechanic   2016.0       Delhi  Automobile & Technology   
87           Spinny   2015.0       Delhi               Automobile   
88  Ess Kay Fincorp      NaN   Rajasthan                  Banking   

                                         What it does  \
0          Provides a range of male grooming products   
1   A learning platform that provides topic-based ...   
2             It aims to make learning fun via games.   
3               Provides interior designing solutions   
4   It is a seed company engaged in production, pr...   
..                                                ...   
84  It connects client requirements to their suppl...   
85                Provides rooms for comfortable stay   
86  Find automobile repair and maintenance service...   
87                                Online car retailer   
88              Organised Non-Banking Finance Company   

                                             Founders  \
0                                  Shantanu Deshpande   
1               Adamas Belva Syah Devara, Iman Usman.   
2                                       Jatin Solanki   
3                        Srikanth Iyer, Rama Harinath   
4                              Narayana Reddy Punyala   
..                                                ...   
84                    Aaditya Sharda, Souvik Sengupta   
85                                     Ritesh Agarwal   
86  Amit Bhasin, Kushal Karwa, Nitin Rana, Rishabh...   
87  Niraj Singh, Ramanshu Mahaur, Ganesh Pawar, Mo...   
88                                     Rajendra Setia   

                                             Investor     Amount($)  \
0                                Sixth Sense Ventures    $6,300,000   
1                                    General Atlantic  $150,000,000   
2      Deepak Parekh, Amitabh Bachchan, Piyush Pandey   $28,000,000   
3   Evolvence India Fund (EIF), Pidilite Group, FJ...   $30,000,000   
4            Innovation in Food and Agriculture (IFA)    $6,000,000   
..                                                ...           ...   
84  Tiger Global, Nexus Venture Partners, Accel Pa...   $20,000,000   
85  MyPreferred Transformation, Avendus Finance, S...  $693,000,000   
86                                    Sequoia Capital    $5,000,000   
87  Norwest Venture Partners, General Catalyst, Fu...   $50,000,000   
88     TPG, Norwest Venture Partners, Evolvence India   $33,000,000   

            Stage  
0             NaN  
1        Series C  
2   Fresh funding  
3        Series D  
4             NaN  
..            ...  
84       Series A  
85            NaN  
86       Series B  
87            NaN  
88            NaN  

[89 rows x 9 columns]

In [4]:
startup_2020


Company/Brand Founded HeadQuarter              Sector  \
0       Aqgromalin    2019     Chennai            AgriTech   
1         Krayonnz    2019   Bangalore              EdTech   
2     PadCare Labs    2018        Pune  Hygiene management   
3            NCOME    2020   New Delhi              Escrow   
4       Gramophone    2016      Indore            AgriTech   
...            ...     ...         ...                 ...   
1050  Leverage Edu     NaN       Delhi              Edtech   
1051         EpiFi     NaN         NaN             Fintech   
1052       Purplle    2012      Mumbai           Cosmetics   
1053        Shuttl    2015       Delhi           Transport   
1054         Pando    2017     Chennai            Logitech   

                                           What it does  \
0                          Cultivating Ideas for Profit   
1     An academy-guardian-scholar centric ecosystem ...   
2      Converting bio-hazardous waste to harmless waste   
3                          Escrow-as-a-service platform   
4     Gramophone is an AgTech platform enabling acce...   
...                                                 ...   
1050  AI enabled marketplace that provides career gu...   
1051  It offers customers with a single interface fo...   
1052         Online makeup and beauty products retailer   
1053                    App based bus aggregator serice   
1054            Networked logistics management software   

                                               Founders  \
0                       Prasanna Manogaran, Bharani C L   
1                      Saurabh Dixit, Gurudutt Upadhyay   
2                                       Ajinkya Dhariya   
3                                         Ritesh Tiwari   
4     Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   
...                                                 ...   
1050                                  Akshay Chaturvedi   
1051                    Sujith Narayanan, Sumit Gwalani   
1052                          Manish Taneja, Rahul Dash   
1053                      Amit Singh, Deepanshu Malviya   
1054                     Jayakrishnan, Abhijeet Manohar   

                                   Investor    Amount($)       Stage  \
0                           Angel investors     $200,000         NaN   
1                           GSF Accelerator     $100,000    Pre-seed   
2                            Venture Center  Undisclosed    Pre-seed   
3       Venture Catalysts, PointOne Capital     $400,000         NaN   
4       Siana Capital Management, Info Edge     $340,000         NaN   
...                                     ...          ...         ...   
1050  DSG Consumer Partners, Blume Ventures   $1,500,000         NaN   
1051          Sequoia India, Ribbit Capital  $13,200,000  Seed Round   
1052                             Verlinvest   $8,000,000         NaN   
1053             SIG Global India Fund LLP.   $8,043,000    Series C   
1054                      Chiratae Ventures   $9,000,000    Series A   

     Unnamed: 9  
0           NaN  
1           NaN  
2           NaN  
3           NaN  
4           NaN  
...         ...  
1050        NaN  
1051        NaN  
1052        NaN  
1053        NaN  
1054        NaN  

[1055 rows x 10 columns]

In [5]:
startup_2021

Company/Brand  Founded HeadQuarter                 Sector  \
0     Unbox Robotics   2019.0   Bangalore             AI startup   
1             upGrad   2015.0      Mumbai                 EdTech   
2        Lead School   2012.0      Mumbai                 EdTech   
3            Bizongo   2015.0      Mumbai         B2B E-commerce   
4           FypMoney   2021.0    Gurugram                FinTech   
...              ...      ...         ...                    ...   
1204        Gigforce   2019.0    Gurugram  Staffing & Recruiting   
1205          Vahdam   2015.0   New Delhi       Food & Beverages   
1206    Leap Finance   2019.0   Bangalore     Financial Services   
1207    CollegeDekho   2015.0    Gurugram                 EdTech   
1208          WeRize   2019.0   Bangalore     Financial Services   

                                           What it does  \
0     Unbox Robotics builds on-demand AI-driven ware...   
1        UpGrad is an online higher education platform.   
2     LEAD School offers technology based school tra...   
3     Bizongo is a business-to-business online marke...   
4     FypMoney is Digital NEO Bank for Teenagers, em...   
...                                                 ...   
1204                  A gig/on-demand staffing company.   
1205  VAHDAM is among the world’s first vertically i...   
1206  International education loans for high potenti...   
1207  Collegedekho.com is Student’s Partner, Friend ...   
1208  India’s first socially distributed full stack ...   

                                               Founders  \
0                           Pramod Ghadge, Shahid Memon   
1     Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                            Smita Deorah, Sumeet Mehta   
3               Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                         Kapil Banwari   
...                                                 ...   
1204                        Chirag Mittal, Anirudh Syal   
1205                                         Bala Sarda   
1206                         Arnav Kumar, Vaibhav Singh   
1207                                       Ruchir Arora   
1208                      Vishal Chopra, Himanshu Gupta   

                                           Investor     Amount($)  \
0                       BEENEXT, Entrepreneur First    $1,200,000   
1          Unilazer Ventures, IIFL Asset Management  $120,000,000   
2                  GSV Ventures, Westbridge Capital   $30,000,000   
3                            CDC Group, IDG Capital   $51,000,000   
4     Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    $2,000,000   
...                                             ...           ...   
1204                                Endiya Partners      $3000000   
1205                                       IIFL AMC     $20000000   
1206                                   Owl Ventures     $55000000   
1207               Winter Capital, ETS, Man Capital     $26000000   
1208                 3one4 Capital, Kalaari Capital      $8000000   

             Stage  
0     Pre-series A  
1              NaN  
2         Series D  
3         Series C  
4             Seed  
...            ...  
1204  Pre-series A  
1205      Series D  
1206      Series C  
1207      Series B  
1208      Series A  

[1209 rows x 9 columns]

### STATING THE HYPOTHESIS

### HO (Null): Sector of service is not dependent on the amount raised.

### H1(Alternative): Sector of service is dependent on the amount raised.

### ANALYTICAL QUESTIONS

### 1) What is the total amount raised by all the investors

### 2) Which sector has the highest amount invested.

### 3) Which year has the most start up founded.

### 4) Which sector has the highest stage of funding.

### 5) Which Company has the highest amount invested on and who is the investor.

### 6) What is the Maximum, Minimum and Average amount raised.

### DATA PRE-PROCESSING

In [6]:
# 2018 STARTUP DATA PROCESSING

startup_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Company Name   526 non-null    object
 1   Industry       526 non-null    object
 2   Round/Series   526 non-null    object
 3   Amount         526 non-null    object
 4   Location       526 non-null    object
 5   About Company  526 non-null    object
dtypes: object(6)
memory usage: 24.8+ KB


In [7]:
startup_2018.describe()

Company Name Industry Round/Series Amount  \
count               526      526          526    526   
unique              525      405           21    198   
top     TheCollegeFever        —         Seed      —   
freq                  2       30          280    148   

                           Location  \
count                           526   
unique                           50   
top     Bangalore, Karnataka, India   
freq                            102   

                                            About Company  
count                                                 526  
unique                                                524  
top     TheCollegeFever is a hub for fun, fiesta and f...  
freq                                                    2

In [8]:
startup_2018.head(10)

Company Name                                           Industry  \
0     TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1     Happy Cow Dairy                               Agriculture, Farming   
2          MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3         PayMe India                        Financial Services, FinTech   
4            Eunimart                 E-Commerce Platforms, Retail, SaaS   
5              Hasura                   Cloud Infrastructure, PaaS, SaaS   
6           Tripshelf                     Internet, Leisure, Marketplace   
7        Hyperdata.IO                                    Market Research   
8        Freightwalla       Information Services, Information Technology   
9  Microchip Payments                                    Mobile Payments   

  Round/Series       Amount                          Location  \
0         Seed       250000       Bangalore, Karnataka, India   
1         Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2     Series A  ₹65,000,000           Gurgaon, Haryana, India   
3        Angel      2000000       Noida, Uttar Pradesh, India   
4         Seed            —  Hyderabad, Andhra Pradesh, India   
5         Seed      1600000       Bengaluru, Karnataka, India   
6         Seed  ₹16,000,000             Kalkaji, Delhi, India   
7        Angel  ₹50,000,000  Hyderabad, Andhra Pradesh, India   
8         Seed            —        Mumbai, Maharashtra, India   
9         Seed            —       Bangalore, Karnataka, India   

                                       About Company  
0  TheCollegeFever is a hub for fun, fiesta and f...  
1  A startup which aggregates milk from dairy far...  
2          Leading Online Loans Marketplace in India  
3  PayMe India is an innovative FinTech organizat...  
4  Eunimart is a one stop solution for merchants ...  
5  Hasura is a platform that allows developers to...  
6  Tripshelf is an online market place for holida...  
7  Hyperdata combines advanced machine learning w...  
8  Freightwalla is an international forwarder tha...  
9  Microchip payments is a mobile-based payment a...

In [9]:
startup_2018.shape

(526, 6)

In [10]:
startup_2018.dtypes

Company Name     object
Industry         object
Round/Series     object
Amount           object
Location         object
About Company    object
dtype: object

In [11]:
startup_2018.isna().sum()

Company Name     0
Industry         0
Round/Series     0
Amount           0
Location         0
About Company    0
dtype: int64

### RENAMING THE COLUMNS

In [12]:
startup_2018.rename(columns={'Company Name':'Company/Brand', 'Industry':'Sector', 'Round/Series':'Stage', 'Amount':'Amount ($)',
                             'Location':'HeadQuarter', 'About Company':'What it does'}, inplace = True)

In [13]:
startup_2018

Company/Brand                                             Sector  \
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

        Stage   Amount ($)                       HeadQuarter  \
0        Seed       250000       Bangalore, Karnataka, India   
1        Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2    Series A  ₹65,000,000           Gurgaon, Haryana, India   
3       Angel      2000000       Noida, Uttar Pradesh, India   
4        Seed            —  Hyderabad, Andhra Pradesh, India   
..        ...          ...                               ...   
521  Series C    225000000       Bangalore, Karnataka, India   
522  Series A            —           Haryana, Haryana, India   
523      Seed         7500        Mumbai, Maharashtra, India   
524      Seed  ₹35,000,000        Mumbai, Maharashtra, India   
525  Series C     35000000        Chennai, Tamil Nadu, India   

                                          What it does  
0    TheCollegeFever is a hub for fun, fiesta and f...  
1    A startup which aggregates milk from dairy far...  
2            Leading Online Loans Marketplace in India  
3    PayMe India is an innovative FinTech organizat...  
4    Eunimart is a one stop solution for merchants ...  
..                                                 ...  
521  Udaan is a B2B trade platform, designed specif...  
522            HappyEasyGo is an online travel domain.  
523  Mombay is a unique opportunity for housewives ...  
524  Droni Tech manufacture UAVs and develop softwa...  
525       Welcome to India's most convenient pharmacy!  

[526 rows x 6 columns]

### FEATURE ENGINEERING 

###               CREATING THE YEAR OF FUNDING COLUMN

In [14]:
startup_2018['Year of Funding'] = '2018'

In [15]:
startup_2018

Company/Brand                                             Sector  \
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

        Stage   Amount ($)                       HeadQuarter  \
0        Seed       250000       Bangalore, Karnataka, India   
1        Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2    Series A  ₹65,000,000           Gurgaon, Haryana, India   
3       Angel      2000000       Noida, Uttar Pradesh, India   
4        Seed            —  Hyderabad, Andhra Pradesh, India   
..        ...          ...                               ...   
521  Series C    225000000       Bangalore, Karnataka, India   
522  Series A            —           Haryana, Haryana, India   
523      Seed         7500        Mumbai, Maharashtra, India   
524      Seed  ₹35,000,000        Mumbai, Maharashtra, India   
525  Series C     35000000        Chennai, Tamil Nadu, India   

                                          What it does Year of Funding  
0    TheCollegeFever is a hub for fun, fiesta and f...            2018  
1    A startup which aggregates milk from dairy far...            2018  
2            Leading Online Loans Marketplace in India            2018  
3    PayMe India is an innovative FinTech organizat...            2018  
4    Eunimart is a one stop solution for merchants ...            2018  
..                                                 ...             ...  
521  Udaan is a B2B trade platform, designed specif...            2018  
522            HappyEasyGo is an online travel domain.            2018  
523  Mombay is a unique opportunity for housewives ...            2018  
524  Droni Tech manufacture UAVs and develop softwa...            2018  
525       Welcome to India's most convenient pharmacy!            2018  

[526 rows x 7 columns]

### CREATING ANOTHER AMOUNT (USD) COLUMN

### THIS COLUMN WILL BE USED TO ISOLATE THE RUPEES, DOLLARS AND EMPTY SIGNS

In [16]:
# CREATING THE AMOUNT (USD) COLUMN

startup_2018['Amount(USD)'] = ' '


# USING THE STRING FUNCTION

inr = startup_2018[startup_2018["Amount ($)"].str.startswith('₹')]

inr


Company/Brand                                             Sector  \
1    Happy Cow Dairy                               Agriculture, Farming   
2         MyLoanCare   Credit, Financial Services, Lending, Marketplace   
6          Tripshelf                     Internet, Leisure, Marketplace   
7       Hyperdata.IO                                    Market Research   
15           Pitstop        Automotive, Search Engine, Service Industry   
..               ...                                                ...   
513            Nykaa                          Beauty, Fashion, Wellness   
514          Chaayos                Food and Beverage, Restaurants, Tea   
516         LT Foods  Food and Beverage, Food Processing, Manufacturing   
517       Multibashi                               E-Learning, Internet   
524       Droni Tech                             Information Technology   

                Stage      Amount ($)                       HeadQuarter  \
1                Seed     ₹40,000,000        Mumbai, Maharashtra, India   
2            Series A     ₹65,000,000           Gurgaon, Haryana, India   
6                Seed     ₹16,000,000             Kalkaji, Delhi, India   
7               Angel     ₹50,000,000  Hyderabad, Andhra Pradesh, India   
15               Seed    ₹100,000,000       Bengaluru, Karnataka, India   
..                ...             ...                               ...   
513  Secondary Market  ₹1,130,000,000        Mumbai, Maharashtra, India   
514          Series B    ₹810,000,000           New Delhi, Delhi, India   
516   Post-IPO Equity  ₹1,400,000,000           New Delhi, Delhi, India   
517              Seed     ₹10,000,000       Bengaluru, Karnataka, India   
524              Seed     ₹35,000,000        Mumbai, Maharashtra, India   

                                          What it does Year of Funding  \
1    A startup which aggregates milk from dairy far...            2018   
2            Leading Online Loans Marketplace in India            2018   
6    Tripshelf is an online market place for holida...            2018   
7    Hyperdata combines advanced machine learning w...            2018   
15   Pitstop offers general repair and maintenance ...            2018   
..                                                 ...             ...   
513  Nykaa.com is a premier online beauty and welln...            2018   
514  Chaayos was born in November 2012 out of this ...            2018   
516  LT Foods believe that nature will continue to ...            2018   
517                   Free language learning platform.            2018   
524  Droni Tech manufacture UAVs and develop softwa...            2018   

    Amount(USD)  
1                
2                
6                
7                
15               
..          ...  
513              
514              
516              
517              
524              

[144 rows x 8 columns]

### FILLING THE AMOUNT (USD) COLUMN WITH INR FOR ₹, USD FOR $ AND EMPTY FOR -


In [17]:
startup_2018.loc[startup_2018['Amount ($)'].str.startswith('₹'), 'Amount(USD)'] = 'INR'

startup_2018.loc[startup_2018['Amount ($)'].str.startswith('$'), 'Amount(USD)'] = ' '

startup_2018.loc[startup_2018['Amount ($)'] =='—', 'Amount(USD)'] = 'NUL'



startup_2018

Company/Brand                                             Sector  \
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

        Stage   Amount ($)                       HeadQuarter  \
0        Seed       250000       Bangalore, Karnataka, India   
1        Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2    Series A  ₹65,000,000           Gurgaon, Haryana, India   
3       Angel      2000000       Noida, Uttar Pradesh, India   
4        Seed            —  Hyderabad, Andhra Pradesh, India   
..        ...          ...                               ...   
521  Series C    225000000       Bangalore, Karnataka, India   
522  Series A            —           Haryana, Haryana, India   
523      Seed         7500        Mumbai, Maharashtra, India   
524      Seed  ₹35,000,000        Mumbai, Maharashtra, India   
525  Series C     35000000        Chennai, Tamil Nadu, India   

                                          What it does Year of Funding  \
0    TheCollegeFever is a hub for fun, fiesta and f...            2018   
1    A startup which aggregates milk from dairy far...            2018   
2            Leading Online Loans Marketplace in India            2018   
3    PayMe India is an innovative FinTech organizat...            2018   
4    Eunimart is a one stop solution for merchants ...            2018   
..                                                 ...             ...   
521  Udaan is a B2B trade platform, designed specif...            2018   
522            HappyEasyGo is an online travel domain.            2018   
523  Mombay is a unique opportunity for housewives ...            2018   
524  Droni Tech manufacture UAVs and develop softwa...            2018   
525       Welcome to India's most convenient pharmacy!            2018   

    Amount(USD)  
0                
1           INR  
2           INR  
3                
4           NUL  
..          ...  
521              
522         NUL  
523              
524         INR  
525              

[526 rows x 8 columns]

### REMOVING THE DOLLAR AND RUPEES SIGN 

### AND REMOVING ALL COMAS, HIPHENS IN THE AMOUNT ($) COLUMN

In [18]:
startup_2018.loc[startup_2018['Amount ($)'].str.startswith('₹'), 'Amount ($)'] = startup_2018['Amount ($)'].str[1:]

startup_2018.loc[startup_2018['Amount ($)'].str.startswith('$'), 'Amount ($)'] = startup_2018['Amount ($)'].str[1:]

startup_2018.loc[startup_2018['Amount ($)'].str.startswith('—'), 'Amount ($)'] = '0.0'

startup_2018.loc[startup_2018['Amount ($)'].str.contains(',', regex=True), 'Amount ($)'] = startup_2018['Amount ($)'].str.replace(',',' ')


startup_2018

Company/Brand                                             Sector  \
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

        Stage  Amount ($)                       HeadQuarter  \
0        Seed      250000       Bangalore, Karnataka, India   
1        Seed  40 000 000        Mumbai, Maharashtra, India   
2    Series A  65 000 000           Gurgaon, Haryana, India   
3       Angel     2000000       Noida, Uttar Pradesh, India   
4        Seed         0.0  Hyderabad, Andhra Pradesh, India   
..        ...         ...                               ...   
521  Series C   225000000       Bangalore, Karnataka, India   
522  Series A         0.0           Haryana, Haryana, India   
523      Seed        7500        Mumbai, Maharashtra, India   
524      Seed  35 000 000        Mumbai, Maharashtra, India   
525  Series C    35000000        Chennai, Tamil Nadu, India   

                                          What it does Year of Funding  \
0    TheCollegeFever is a hub for fun, fiesta and f...            2018   
1    A startup which aggregates milk from dairy far...            2018   
2            Leading Online Loans Marketplace in India            2018   
3    PayMe India is an innovative FinTech organizat...            2018   
4    Eunimart is a one stop solution for merchants ...            2018   
..                                                 ...             ...   
521  Udaan is a B2B trade platform, designed specif...            2018   
522            HappyEasyGo is an online travel domain.            2018   
523  Mombay is a unique opportunity for housewives ...            2018   
524  Droni Tech manufacture UAVs and develop softwa...            2018   
525       Welcome to India's most convenient pharmacy!            2018   

    Amount(USD)  
0                
1           INR  
2           INR  
3                
4           NUL  
..          ...  
521              
522         NUL  
523              
524         INR  
525              

[526 rows x 8 columns]

In [19]:
# checking datatype

startup_2018.dtypes

Company/Brand      object
Sector             object
Stage              object
Amount ($)         object
HeadQuarter        object
What it does       object
Year of Funding    object
Amount(USD)        object
dtype: object

### CONVERTING AMOUNT ($) AND YEAR OF FUNDING DATATYPE FROM OBJECT TO FLOAT AND INT

In [20]:
#startup_2018['Amount ($)'] = startup_2018['Amount ($)'].astype(float)

startup_2018['Amount ($)'] = pd.to_numeric(startup_2018['Amount ($)'], errors='coerce')

startup_2018['Year of Funding'] = pd.to_numeric(startup_2018['Year of Funding'], errors='coerce')

print(startup_2018.dtypes)

Company/Brand       object
Sector              object
Stage               object
Amount ($)         float64
HeadQuarter         object
What it does        object
Year of Funding      int64
Amount(USD)         object
dtype: object


### CURRENCY CONVERSION - WE ARE USING 68.41 RUPEES TO 1 USD

In [21]:
# indian exchange rate = 68.41

indian_rate = 68.41

indian_rate

68.41

In [22]:
startup_2018.loc[startup_2018['Amount(USD)'] == 'INR', 'Amount ($)'] = startup_2018['Amount ($)']/indian_rate

startup_2018

Company/Brand                                             Sector  \
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

        Stage   Amount ($)                       HeadQuarter  \
0        Seed     250000.0       Bangalore, Karnataka, India   
1        Seed          NaN        Mumbai, Maharashtra, India   
2    Series A          NaN           Gurgaon, Haryana, India   
3       Angel    2000000.0       Noida, Uttar Pradesh, India   
4        Seed          0.0  Hyderabad, Andhra Pradesh, India   
..        ...          ...                               ...   
521  Series C  225000000.0       Bangalore, Karnataka, India   
522  Series A          0.0           Haryana, Haryana, India   
523      Seed       7500.0        Mumbai, Maharashtra, India   
524      Seed          NaN        Mumbai, Maharashtra, India   
525  Series C   35000000.0        Chennai, Tamil Nadu, India   

                                          What it does  Year of Funding  \
0    TheCollegeFever is a hub for fun, fiesta and f...             2018   
1    A startup which aggregates milk from dairy far...             2018   
2            Leading Online Loans Marketplace in India             2018   
3    PayMe India is an innovative FinTech organizat...             2018   
4    Eunimart is a one stop solution for merchants ...             2018   
..                                                 ...              ...   
521  Udaan is a B2B trade platform, designed specif...             2018   
522            HappyEasyGo is an online travel domain.             2018   
523  Mombay is a unique opportunity for housewives ...             2018   
524  Droni Tech manufacture UAVs and develop softwa...             2018   
525       Welcome to India's most convenient pharmacy!             2018   

    Amount(USD)  
0                
1           INR  
2           INR  
3                
4           NUL  
..          ...  
521              
522         NUL  
523              
524         INR  
525              

[526 rows x 8 columns]

In [23]:
startup_2018.head(100)

Company/Brand                                             Sector  \
0          TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1          Happy Cow Dairy                               Agriculture, Farming   
2               MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3              PayMe India                        Financial Services, FinTech   
4                 Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                     ...                                                ...   
95              AuthMetrik  B2B, Biometrics, Cyber Security, Fraud Detecti...   
96                  Khidki                    Artificial Intelligence, Social   
97           LetsTransport                  Logistics, Transportation, Travel   
98  Next Digital Solutions       Digital Marketing, SEM, SEO, Web Development   
99          Portea Medical                     Health Care, Hospital, Medical   

             Stage  Amount ($)                       HeadQuarter  \
0             Seed    250000.0       Bangalore, Karnataka, India   
1             Seed         NaN        Mumbai, Maharashtra, India   
2         Series A         NaN           Gurgaon, Haryana, India   
3            Angel   2000000.0       Noida, Uttar Pradesh, India   
4             Seed         0.0  Hyderabad, Andhra Pradesh, India   
..             ...         ...                               ...   
95           Grant         NaN           Gurgaon, Haryana, India   
96            Seed         0.0       Bangalore, Karnataka, India   
97        Series B         NaN       Bangalore, Karnataka, India   
98           Angel         NaN            Kota, Rajasthan, India   
99  Debt Financing         NaN       Bangalore, Karnataka, India   

                                         What it does  Year of Funding  \
0   TheCollegeFever is a hub for fun, fiesta and f...             2018   
1   A startup which aggregates milk from dairy far...             2018   
2           Leading Online Loans Marketplace in India             2018   
3   PayMe India is an innovative FinTech organizat...             2018   
4   Eunimart is a one stop solution for merchants ...             2018   
..                                                ...              ...   
95  SaaS, B2B, Security, Stop account sharing, Fra...             2018   
96  Vernacular Social Network Focused on Town Leve...             2018   
97   Lets transport is a logistics solution provider.             2018   
98  Next Digital Solutions is website design & Dig...             2018   
99  Portea Medical is the largest and fastest grow...             2018   

   Amount(USD)  
0               
1          INR  
2          INR  
3               
4          NUL  
..         ...  
95              
96         NUL  
97         INR  
98         INR  
99         INR  

[100 rows x 8 columns]

In [24]:
startup_2018['Amount ($)'].describe()

count    3.230000e+02
mean     6.609109e+06
std      3.063988e+07
min      0.000000e+00
25%      0.000000e+00
50%      1.500000e+05
75%      1.500000e+06
max      3.650000e+08
Name: Amount ($), dtype: float64

### REMOVING THE REGION AND COUNTRY FROM THE HEADQUARTER LEAVING ONLY THE CITY TO ALIGN WITH THE REST OF THE DATA 

In [25]:
startup_2018['HeadQuarter'] = startup_2018['HeadQuarter'].str.split(',').str[0]

startup_2018

Company/Brand                                             Sector  \
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

        Stage   Amount ($) HeadQuarter  \
0        Seed     250000.0   Bangalore   
1        Seed          NaN      Mumbai   
2    Series A          NaN     Gurgaon   
3       Angel    2000000.0       Noida   
4        Seed          0.0   Hyderabad   
..        ...          ...         ...   
521  Series C  225000000.0   Bangalore   
522  Series A          0.0     Haryana   
523      Seed       7500.0      Mumbai   
524      Seed          NaN      Mumbai   
525  Series C   35000000.0     Chennai   

                                          What it does  Year of Funding  \
0    TheCollegeFever is a hub for fun, fiesta and f...             2018   
1    A startup which aggregates milk from dairy far...             2018   
2            Leading Online Loans Marketplace in India             2018   
3    PayMe India is an innovative FinTech organizat...             2018   
4    Eunimart is a one stop solution for merchants ...             2018   
..                                                 ...              ...   
521  Udaan is a B2B trade platform, designed specif...             2018   
522            HappyEasyGo is an online travel domain.             2018   
523  Mombay is a unique opportunity for housewives ...             2018   
524  Droni Tech manufacture UAVs and develop softwa...             2018   
525       Welcome to India's most convenient pharmacy!             2018   

    Amount(USD)  
0                
1           INR  
2           INR  
3                
4           NUL  
..          ...  
521              
522         NUL  
523              
524         INR  
525              

[526 rows x 8 columns]

### REPLACING MIS-SPELT BENGALURU TO BANGALORE IN HEADQUARTER COLUMN

In [26]:
Bengaluru = startup_2018.loc[startup_2018['HeadQuarter'] == 'HeadQuarter'].count()
if(Bengaluru['HeadQuarter'].sum() > 0):
    startup_2018['HeadQuarter'] = startup_2018['HeadQuarter'].str.replace('Bengaluru', 'Bangalore')

startup_2018

Company/Brand                                             Sector  \
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

        Stage   Amount ($) HeadQuarter  \
0        Seed     250000.0   Bangalore   
1        Seed          NaN      Mumbai   
2    Series A          NaN     Gurgaon   
3       Angel    2000000.0       Noida   
4        Seed          0.0   Hyderabad   
..        ...          ...         ...   
521  Series C  225000000.0   Bangalore   
522  Series A          0.0     Haryana   
523      Seed       7500.0      Mumbai   
524      Seed          NaN      Mumbai   
525  Series C   35000000.0     Chennai   

                                          What it does  Year of Funding  \
0    TheCollegeFever is a hub for fun, fiesta and f...             2018   
1    A startup which aggregates milk from dairy far...             2018   
2            Leading Online Loans Marketplace in India             2018   
3    PayMe India is an innovative FinTech organizat...             2018   
4    Eunimart is a one stop solution for merchants ...             2018   
..                                                 ...              ...   
521  Udaan is a B2B trade platform, designed specif...             2018   
522            HappyEasyGo is an online travel domain.             2018   
523  Mombay is a unique opportunity for housewives ...             2018   
524  Droni Tech manufacture UAVs and develop softwa...             2018   
525       Welcome to India's most convenient pharmacy!             2018   

    Amount(USD)  
0                
1           INR  
2           INR  
3                
4           NUL  
..          ...  
521              
522         NUL  
523              
524         INR  
525              

[526 rows x 8 columns]

In [27]:
#DROPPING AMOUNT(USD) COLUMN

startup_2018.drop(['Amount(USD)'],  axis=1)

Company/Brand                                             Sector  \
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

        Stage   Amount ($) HeadQuarter  \
0        Seed     250000.0   Bangalore   
1        Seed          NaN      Mumbai   
2    Series A          NaN     Gurgaon   
3       Angel    2000000.0       Noida   
4        Seed          0.0   Hyderabad   
..        ...          ...         ...   
521  Series C  225000000.0   Bangalore   
522  Series A          0.0     Haryana   
523      Seed       7500.0      Mumbai   
524      Seed          NaN      Mumbai   
525  Series C   35000000.0     Chennai   

                                          What it does  Year of Funding  
0    TheCollegeFever is a hub for fun, fiesta and f...             2018  
1    A startup which aggregates milk from dairy far...             2018  
2            Leading Online Loans Marketplace in India             2018  
3    PayMe India is an innovative FinTech organizat...             2018  
4    Eunimart is a one stop solution for merchants ...             2018  
..                                                 ...              ...  
521  Udaan is a B2B trade platform, designed specif...             2018  
522            HappyEasyGo is an online travel domain.             2018  
523  Mombay is a unique opportunity for housewives ...             2018  
524  Droni Tech manufacture UAVs and develop softwa...             2018  
525       Welcome to India's most convenient pharmacy!             2018  

[526 rows x 7 columns]

### SAVING CLEANED 2018 DATASET

In [28]:
startup_2018.to_csv('cleaned_2018.csv', index = False)

### PROCESSING 2020 DATASET

In [29]:
startup_2020

Company/Brand Founded HeadQuarter              Sector  \
0       Aqgromalin    2019     Chennai            AgriTech   
1         Krayonnz    2019   Bangalore              EdTech   
2     PadCare Labs    2018        Pune  Hygiene management   
3            NCOME    2020   New Delhi              Escrow   
4       Gramophone    2016      Indore            AgriTech   
...            ...     ...         ...                 ...   
1050  Leverage Edu     NaN       Delhi              Edtech   
1051         EpiFi     NaN         NaN             Fintech   
1052       Purplle    2012      Mumbai           Cosmetics   
1053        Shuttl    2015       Delhi           Transport   
1054         Pando    2017     Chennai            Logitech   

                                           What it does  \
0                          Cultivating Ideas for Profit   
1     An academy-guardian-scholar centric ecosystem ...   
2      Converting bio-hazardous waste to harmless waste   
3                          Escrow-as-a-service platform   
4     Gramophone is an AgTech platform enabling acce...   
...                                                 ...   
1050  AI enabled marketplace that provides career gu...   
1051  It offers customers with a single interface fo...   
1052         Online makeup and beauty products retailer   
1053                    App based bus aggregator serice   
1054            Networked logistics management software   

                                               Founders  \
0                       Prasanna Manogaran, Bharani C L   
1                      Saurabh Dixit, Gurudutt Upadhyay   
2                                       Ajinkya Dhariya   
3                                         Ritesh Tiwari   
4     Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   
...                                                 ...   
1050                                  Akshay Chaturvedi   
1051                    Sujith Narayanan, Sumit Gwalani   
1052                          Manish Taneja, Rahul Dash   
1053                      Amit Singh, Deepanshu Malviya   
1054                     Jayakrishnan, Abhijeet Manohar   

                                   Investor    Amount($)       Stage  \
0                           Angel investors     $200,000         NaN   
1                           GSF Accelerator     $100,000    Pre-seed   
2                            Venture Center  Undisclosed    Pre-seed   
3       Venture Catalysts, PointOne Capital     $400,000         NaN   
4       Siana Capital Management, Info Edge     $340,000         NaN   
...                                     ...          ...         ...   
1050  DSG Consumer Partners, Blume Ventures   $1,500,000         NaN   
1051          Sequoia India, Ribbit Capital  $13,200,000  Seed Round   
1052                             Verlinvest   $8,000,000         NaN   
1053             SIG Global India Fund LLP.   $8,043,000    Series C   
1054                      Chiratae Ventures   $9,000,000    Series A   

     Unnamed: 9  
0           NaN  
1           NaN  
2           NaN  
3           NaN  
4           NaN  
...         ...  
1050        NaN  
1051        NaN  
1052        NaN  
1053        NaN  
1054        NaN  

[1055 rows x 10 columns]

In [30]:
startup_2020.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1055 entries, 0 to 1054
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Company/Brand  1055 non-null   object
 1   Founded        843 non-null    object
 2   HeadQuarter    961 non-null    object
 3   Sector         1042 non-null   object
 4   What it does   1055 non-null   object
 5   Founders       1043 non-null   object
 6   Investor       1017 non-null   object
 7   Amount($)      1052 non-null   object
 8   Stage          591 non-null    object
 9   Unnamed: 9     2 non-null      object
dtypes: object(10)
memory usage: 82.5+ KB


In [31]:
startup_2020.head()

Company/Brand Founded HeadQuarter              Sector  \
0    Aqgromalin    2019     Chennai            AgriTech   
1      Krayonnz    2019   Bangalore              EdTech   
2  PadCare Labs    2018        Pune  Hygiene management   
3         NCOME    2020   New Delhi              Escrow   
4    Gramophone    2016      Indore            AgriTech   

                                        What it does  \
0                       Cultivating Ideas for Profit   
1  An academy-guardian-scholar centric ecosystem ...   
2   Converting bio-hazardous waste to harmless waste   
3                       Escrow-as-a-service platform   
4  Gramophone is an AgTech platform enabling acce...   

                                            Founders  \
0                    Prasanna Manogaran, Bharani C L   
1                   Saurabh Dixit, Gurudutt Upadhyay   
2                                    Ajinkya Dhariya   
3                                      Ritesh Tiwari   
4  Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   

                              Investor    Amount($)     Stage Unnamed: 9  
0                      Angel investors     $200,000       NaN        NaN  
1                      GSF Accelerator     $100,000  Pre-seed        NaN  
2                       Venture Center  Undisclosed  Pre-seed        NaN  
3  Venture Catalysts, PointOne Capital     $400,000       NaN        NaN  
4  Siana Capital Management, Info Edge     $340,000       NaN        NaN

In [32]:
startup_2020.isna().sum()

Company/Brand       0
Founded           212
HeadQuarter        94
Sector             13
What it does        0
Founders           12
Investor           38
Amount($)           3
Stage             464
Unnamed: 9       1053
dtype: int64

In [33]:
startup_2020.shape

(1055, 10)

### WE ARE GOING TO CLEAN THIS DATA COLUMN BY COLUMN

### CLEANING COMPANY/BRAND COLUMN

In [34]:
startup_2020['Company/Brand'].isna().sum()

0

In [35]:
startup_2020['Company/Brand']

0         Aqgromalin
1           Krayonnz
2       PadCare Labs
3              NCOME
4         Gramophone
            ...     
1050    Leverage Edu
1051           EpiFi
1052         Purplle
1053          Shuttl
1054           Pando
Name: Company/Brand, Length: 1055, dtype: object

### THERE IS NO MISSING VALUES IN THE COMPANY/BRAND COLUMN

### CLEANING THE 'FOUNDED ' COLUMN

In [36]:
startup_2020["Founded"].isnull().sum()

212

### There are 212 empty values. 

### We will be using  mode or mean of the column to fill the missing values. 

In [37]:
# copying the column

founded_2020 = startup_2020.Founded 

founded_2020

0       2019
1       2019
2       2018
3       2020
4       2016
        ... 
1050     NaN
1051     NaN
1052    2012
1053    2015
1054    2017
Name: Founded, Length: 1055, dtype: object

### From above the it has "-" and "nan" values. They have to be dropped inorder to find mean and mode.

In [38]:
# searching for the index of the value "-" from the column
for i in range(startup_2020.shape[0]):
    if founded_2020[i] == "-":
        print(i)

482


### It has a Total number of 482 "-"and "nan" values that need to be dropped

In [39]:
founded_2020 = founded_2020.drop(482)

In [40]:
founded_2020 = founded_2020.dropna()

founded_2020

0       2019
1       2019
2       2018
3       2020
4       2016
        ... 
1048    2016
1049    2017
1052    2012
1053    2015
1054    2017
Name: Founded, Length: 842, dtype: object

In [41]:
founded_2020.unique()

array(['2019', '2018', '2020', '2016', '2008', '2015', '2017', '2014',
       '1998', '2007', '2011', '1982', '2013', '2009', '2012', '1995',
       '2010', '2006', '1978', '1999', '1994', '2005', '1973', '2002',
       '2004', '2001'], dtype=object)

### Founded_2020 is free from error for our calculation. 

### Let us convert it to integer

In [42]:
# converting to int data type

founded_2020 = founded_2020.astype(int) 

founded_2020

0       2019
1       2019
2       2018
3       2020
4       2016
        ... 
1048    2016
1049    2017
1052    2012
1053    2015
1054    2017
Name: Founded, Length: 842, dtype: int32

In [43]:
founded_2020.info()

<class 'pandas.core.series.Series'>
Int64Index: 842 entries, 0 to 1054
Series name: Founded
Non-Null Count  Dtype
--------------  -----
842 non-null    int32
dtypes: int32(1)
memory usage: 9.9 KB


### Calculating Mode and Mean

In [44]:
founded_2020.mode()

0    2015
Name: Founded, dtype: int32

In [45]:
founded_2020.mean()

2015.3634204275534

### The Average is 2015. 

###  2015  will be used as an imputer for missed value for column Founded.

In [46]:
startup_2020.Founded.unique()

array(['2019', '2018', '2020', '2016', '2008', '2015', '2017', '2014',
       '1998', '2007', '2011', '1982', '2013', '2009', '2012', '1995',
       '2010', '2006', '1978', nan, '1999', '1994', '2005', '1973', '-',
       '2002', '2004', '2001'], dtype=object)

In [47]:
# repalcing "-" with 2015

startup_2020.Founded = startup_2020.Founded.replace("-",2015) 


# replacing NaN 

startup_2020.Founded = startup_2020.Founded.fillna(2015) 

In [48]:
startup_2020.Founded.isnull().any()

False

### NOW, OUR "FOUNDED " COLUMN IS CLEANED

### CLEANING THE SECTOR COLUMN

In [49]:
# checking if there is MISING value

startup_2020.Sector.isnull().sum() 

13

###  WE HAVE 13 MISSING VALUES INTHE SECTOR COLUMN

In [50]:
# copying the column

sector_2020 = startup_2020.Sector.isnull() 

In [51]:
# finding frequented value

mode = startup_2020.Sector.mode() 

mode

0    Fintech
Name: Sector, dtype: object

In [52]:
 # searching for empty/nan value

for i in range(len(sector_2020)):
    if sector_2020[i] == True:
        startup_2020.Sector[i] = mode

In [53]:
startup_2020.Sector.isnull().any()

False

### SECTOR COLUMN IS NOW CLEANED

### CLEANING INVESTOR COLUMN

In [54]:
# CHECKING FOR MISSING VALUES IN INVESTOR COLUMN

startup_2020.Investor.isnull().sum()

38

### It has a total of 38 missing values

In [55]:
#FILLING THE MISSING VALUES WITH " UNDEFINED"


investor_2020 = startup_2020.Investor.isnull()

In [56]:
for i in range(len(investor_2020)):
    if investor_2020[i] == True:
        startup_2020.Investor[i] = "undefined"

In [57]:
startup_2020.Investor.isnull().sum()

0

### THE INVESTOR COLUMN IS CLEANED NOW

### CLEANING THE AMOUNT COLUMN

In [58]:
startup_2020["Amount($)"].unique()

array(['$200,000', '$100,000', 'Undisclosed', '$400,000', '$340,000',
       '$600,000', '$45,000,000', '$1,000,000', '$2,000,000',
       '$1,200,000', '$660,000,000', '$120,000', '$7,500,000',
       '$5,000,000', '$500,000', '$3,000,000', '$10,000,000',
       '$145,000,000', '$100,000,000', nan, '$21,000,000', '$4,000,000',
       '$20,000,000', '$560,000', '$275,000', '$4,500,000', '$15,000,000',
       '$390,000,000', '$7,000,000', '$5,100,000', '$700,000,000',
       '$2,300,000', '$700,000', '$19,000,000', '$9,000,000',
       '$40,000,000', '$750,000', '$1,500,000', '$7,800,000',
       '$50,000,000', '$80,000,000', '$30,000,000', '$1,700,000',
       '$2,500,000', '$40,000', '$33,000,000', '$35,000,000', '$300,000',
       '$25,000,000', '$3,500,000', '$200,000,000', '$6,000,000',
       '$1,300,000', '$4,100,000', '$575,000', '$800,000', '$28,000,000',
       '$18,000,000', '$3,200,000', '$900,000', '$250,000', '$4,700,000',
       '$75,000,000', '$8,000,000', '$121,000,000'

### FROM THE ABOVE, THE COLUMN HAS "NAN" AND "UNDISCLOSED" VALUES WHICH NEED TO TREATED

### WE ARE GOING TO TREAT THEM BY REPLACING THEM WITH THEIR MEAN VALUES

In [59]:
# copying the column

amount_2020 = startup_2020["Amount($)"] 

In [60]:
# replacing values

amount_2020 = amount_2020.apply(lambda x: str(x).replace(",",""))
amount_2020 = amount_2020.apply(lambda x: str(x).replace("Undisclosed","0"))
amount_2020 = amount_2020.apply(lambda x: str(x).replace("$",""))
amount_2020 = amount_2020.apply(lambda x: str(x).replace("Undislosed","0"))
amount_2020 = amount_2020.apply(lambda x: str(x).replace("Undiclsosed","0"))
amount_2020 = amount_2020.apply(lambda x: str(x).replace("nan","0"))



In [61]:
amount_2020.unique()

array(['200000', '100000', '0', '400000', '340000', '600000', '45000000',
       '1000000', '2000000', '1200000', '660000000', '120000', '7500000',
       '5000000', '500000', '3000000', '10000000', '145000000',
       '100000000', '21000000', '4000000', '20000000', '560000', '275000',
       '4500000', '15000000', '390000000', '7000000', '5100000',
       '700000000', '2300000', '700000', '19000000', '9000000',
       '40000000', '750000', '1500000', '7800000', '50000000', '80000000',
       '30000000', '1700000', '2500000', '40000', '33000000', '35000000',
       '300000', '25000000', '3500000', '200000000', '6000000', '1300000',
       '4100000', '575000', '800000', '28000000', '18000000', '3200000',
       '900000', '250000', '4700000', '75000000', '8000000', '121000000',
       '55000000', '3300000', '11000000', '16000000', '5400000',
       '150000000', '4200000', '22000000', '52000000', '1100000',
       '118000000', '1600000', '18500000', '70000000000', '800000000',
       '400

In [62]:
# eliminating space

data = amount_2020.str.endswith(" ")

for i in range(len(amount_2020)):
    if data[i]:
        amount_2020[i] = amount_2020[i][:-1]
        


In [63]:
# there are two columns having two values

val = {}
for i in range(len(amount_2020)):
    if " " in amount_2020[i]:
        val[i] = amount_2020[i]
val

{465: '887000 23000000', 472: '800000000 to 850000000', 548: ' 8700000'}

In [64]:
# a. replacing values with their mean

val2 = val[465].split(" ")
val2

a = float(val2[0])
b = float(val2[1])

mean = (a + b) / 2
amount_2020[465] = mean



In [65]:
# b. replacing values with their mean

val3 = val[472].split(" to ")
val3

c = float(val2[0])
d = float(val2[1])

mean2 = (c + d) / 2
amount_2020[472] = mean2

In [66]:
# converting to float

amount_2020 = amount_2020.astype(float) 

In [67]:
amount_2020

0         200000.0
1         100000.0
2              0.0
3         400000.0
4         340000.0
           ...    
1050     1500000.0
1051    13200000.0
1052     8000000.0
1053     8043000.0
1054     9000000.0
Name: Amount($), Length: 1055, dtype: float64

In [68]:
#FINDING THE MAEN, MODE AND MEDIA

mean_val = amount_2020.mean()

mean_val

85860249.91073933

In [69]:
amount_2020.mode()

0    0.0
Name: Amount($), dtype: float64

In [70]:
amount_2020.median()

1403400.0

In [71]:
# replacing 0.0 by mean value

amount_2020 = amount_2020.replace(0.0 ,mean_val) 

amount_2020

0       2.000000e+05
1       1.000000e+05
2       8.586025e+07
3       4.000000e+05
4       3.400000e+05
            ...     
1050    1.500000e+06
1051    1.320000e+07
1052    8.000000e+06
1053    8.043000e+06
1054    9.000000e+06
Name: Amount($), Length: 1055, dtype: float64

In [72]:
 # substituting the columns

startup_2020["Amount($)"] = amount_2020

In [73]:
startup_2020

Company/Brand Founded HeadQuarter              Sector  \
0       Aqgromalin    2019     Chennai            AgriTech   
1         Krayonnz    2019   Bangalore              EdTech   
2     PadCare Labs    2018        Pune  Hygiene management   
3            NCOME    2020   New Delhi              Escrow   
4       Gramophone    2016      Indore            AgriTech   
...            ...     ...         ...                 ...   
1050  Leverage Edu    2015       Delhi              Edtech   
1051         EpiFi    2015         NaN             Fintech   
1052       Purplle    2012      Mumbai           Cosmetics   
1053        Shuttl    2015       Delhi           Transport   
1054         Pando    2017     Chennai            Logitech   

                                           What it does  \
0                          Cultivating Ideas for Profit   
1     An academy-guardian-scholar centric ecosystem ...   
2      Converting bio-hazardous waste to harmless waste   
3                          Escrow-as-a-service platform   
4     Gramophone is an AgTech platform enabling acce...   
...                                                 ...   
1050  AI enabled marketplace that provides career gu...   
1051  It offers customers with a single interface fo...   
1052         Online makeup and beauty products retailer   
1053                    App based bus aggregator serice   
1054            Networked logistics management software   

                                               Founders  \
0                       Prasanna Manogaran, Bharani C L   
1                      Saurabh Dixit, Gurudutt Upadhyay   
2                                       Ajinkya Dhariya   
3                                         Ritesh Tiwari   
4     Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   
...                                                 ...   
1050                                  Akshay Chaturvedi   
1051                    Sujith Narayanan, Sumit Gwalani   
1052                          Manish Taneja, Rahul Dash   
1053                      Amit Singh, Deepanshu Malviya   
1054                     Jayakrishnan, Abhijeet Manohar   

                                   Investor     Amount($)       Stage  \
0                           Angel investors  2.000000e+05         NaN   
1                           GSF Accelerator  1.000000e+05    Pre-seed   
2                            Venture Center  8.586025e+07    Pre-seed   
3       Venture Catalysts, PointOne Capital  4.000000e+05         NaN   
4       Siana Capital Management, Info Edge  3.400000e+05         NaN   
...                                     ...           ...         ...   
1050  DSG Consumer Partners, Blume Ventures  1.500000e+06         NaN   
1051          Sequoia India, Ribbit Capital  1.320000e+07  Seed Round   
1052                             Verlinvest  8.000000e+06         NaN   
1053             SIG Global India Fund LLP.  8.043000e+06    Series C   
1054                      Chiratae Ventures  9.000000e+06    Series A   

     Unnamed: 9  
0           NaN  
1           NaN  
2           NaN  
3           NaN  
4           NaN  
...         ...  
1050        NaN  
1051        NaN  
1052        NaN  
1053        NaN  
1054        NaN  

[1055 rows x 10 columns]

### THE AMOUNT COLUMN IS CLEAN NOW

### LET NOW CLEAN THE FOUNDERS COLUMN

In [74]:
startup_2020.Founders.isnull().sum()

12

### It has 12 null values

In [75]:
# Let replace the null values with "Undefined"

founders_2020 = startup_2020.Founders.isnull()

founders_2020

0       False
1       False
2       False
3       False
4       False
        ...  
1050    False
1051    False
1052    False
1053    False
1054    False
Name: Founders, Length: 1055, dtype: bool

In [76]:
for i in range(len(founders_2020)):
    if founders_2020[i] == True:
        startup_2020.Founders[i] = "Undefined"

C:\Users\User\AppData\Local\Temp\ipykernel_23912\2141025315.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  startup_2020.Founders[i] = "Undefined"


In [77]:
startup_2020.Founders.isnull().sum()

0

### Now, we are go to go as the Founders column is clean now..

### LET CLEAN THE STAGE COLUMN

In [78]:
startup_2020.Stage.unique()

array([nan, 'Pre-seed', 'Seed', 'Pre-series A', 'Pre-series', 'Series C',
       'Series A', 'Series B', 'Debt', 'Pre-series C', 'Pre-series B',
       'Series E', 'Bridge', 'Series D', 'Series B2', 'Series F',
       'Pre- series A', 'Edge', 'Series H', 'Pre-Series B', 'Seed A',
       'Series A-1', 'Seed Funding', 'Pre-Seed', 'Seed round',
       'Pre-seed Round', 'Seed Round & Series A', 'Pre Series A',
       'Pre seed Round', 'Angel Round', 'Pre series A1', 'Series E2',
       'Pre series A', 'Seed Round', 'Bridge Round', 'Pre seed round',
       'Pre series B', 'Pre series C', 'Seed Investment', 'Series D1',
       'Mid series', 'Series C, D', 'Seed funding'], dtype=object)

In [79]:
# Filling 

startup_2020.Stage = startup_2020.Stage.fillna(startup_2020.Stage.mode()[0])

In [80]:
startup_2020

Company/Brand Founded HeadQuarter              Sector  \
0       Aqgromalin    2019     Chennai            AgriTech   
1         Krayonnz    2019   Bangalore              EdTech   
2     PadCare Labs    2018        Pune  Hygiene management   
3            NCOME    2020   New Delhi              Escrow   
4       Gramophone    2016      Indore            AgriTech   
...            ...     ...         ...                 ...   
1050  Leverage Edu    2015       Delhi              Edtech   
1051         EpiFi    2015         NaN             Fintech   
1052       Purplle    2012      Mumbai           Cosmetics   
1053        Shuttl    2015       Delhi           Transport   
1054         Pando    2017     Chennai            Logitech   

                                           What it does  \
0                          Cultivating Ideas for Profit   
1     An academy-guardian-scholar centric ecosystem ...   
2      Converting bio-hazardous waste to harmless waste   
3                          Escrow-as-a-service platform   
4     Gramophone is an AgTech platform enabling acce...   
...                                                 ...   
1050  AI enabled marketplace that provides career gu...   
1051  It offers customers with a single interface fo...   
1052         Online makeup and beauty products retailer   
1053                    App based bus aggregator serice   
1054            Networked logistics management software   

                                               Founders  \
0                       Prasanna Manogaran, Bharani C L   
1                      Saurabh Dixit, Gurudutt Upadhyay   
2                                       Ajinkya Dhariya   
3                                         Ritesh Tiwari   
4     Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   
...                                                 ...   
1050                                  Akshay Chaturvedi   
1051                    Sujith Narayanan, Sumit Gwalani   
1052                          Manish Taneja, Rahul Dash   
1053                      Amit Singh, Deepanshu Malviya   
1054                     Jayakrishnan, Abhijeet Manohar   

                                   Investor     Amount($)       Stage  \
0                           Angel investors  2.000000e+05    Series A   
1                           GSF Accelerator  1.000000e+05    Pre-seed   
2                            Venture Center  8.586025e+07    Pre-seed   
3       Venture Catalysts, PointOne Capital  4.000000e+05    Series A   
4       Siana Capital Management, Info Edge  3.400000e+05    Series A   
...                                     ...           ...         ...   
1050  DSG Consumer Partners, Blume Ventures  1.500000e+06    Series A   
1051          Sequoia India, Ribbit Capital  1.320000e+07  Seed Round   
1052                             Verlinvest  8.000000e+06    Series A   
1053             SIG Global India Fund LLP.  8.043000e+06    Series C   
1054                      Chiratae Ventures  9.000000e+06    Series A   

     Unnamed: 9  
0           NaN  
1           NaN  
2           NaN  
3           NaN  
4           NaN  
...         ...  
1050        NaN  
1051        NaN  
1052        NaN  
1053        NaN  
1054        NaN  

[1055 rows x 10 columns]

In [81]:
startup_2020['Year of Funding'] = '2020'

In [82]:
startup_2020

Company/Brand Founded HeadQuarter              Sector  \
0       Aqgromalin    2019     Chennai            AgriTech   
1         Krayonnz    2019   Bangalore              EdTech   
2     PadCare Labs    2018        Pune  Hygiene management   
3            NCOME    2020   New Delhi              Escrow   
4       Gramophone    2016      Indore            AgriTech   
...            ...     ...         ...                 ...   
1050  Leverage Edu    2015       Delhi              Edtech   
1051         EpiFi    2015         NaN             Fintech   
1052       Purplle    2012      Mumbai           Cosmetics   
1053        Shuttl    2015       Delhi           Transport   
1054         Pando    2017     Chennai            Logitech   

                                           What it does  \
0                          Cultivating Ideas for Profit   
1     An academy-guardian-scholar centric ecosystem ...   
2      Converting bio-hazardous waste to harmless waste   
3                          Escrow-as-a-service platform   
4     Gramophone is an AgTech platform enabling acce...   
...                                                 ...   
1050  AI enabled marketplace that provides career gu...   
1051  It offers customers with a single interface fo...   
1052         Online makeup and beauty products retailer   
1053                    App based bus aggregator serice   
1054            Networked logistics management software   

                                               Founders  \
0                       Prasanna Manogaran, Bharani C L   
1                      Saurabh Dixit, Gurudutt Upadhyay   
2                                       Ajinkya Dhariya   
3                                         Ritesh Tiwari   
4     Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   
...                                                 ...   
1050                                  Akshay Chaturvedi   
1051                    Sujith Narayanan, Sumit Gwalani   
1052                          Manish Taneja, Rahul Dash   
1053                      Amit Singh, Deepanshu Malviya   
1054                     Jayakrishnan, Abhijeet Manohar   

                                   Investor     Amount($)       Stage  \
0                           Angel investors  2.000000e+05    Series A   
1                           GSF Accelerator  1.000000e+05    Pre-seed   
2                            Venture Center  8.586025e+07    Pre-seed   
3       Venture Catalysts, PointOne Capital  4.000000e+05    Series A   
4       Siana Capital Management, Info Edge  3.400000e+05    Series A   
...                                     ...           ...         ...   
1050  DSG Consumer Partners, Blume Ventures  1.500000e+06    Series A   
1051          Sequoia India, Ribbit Capital  1.320000e+07  Seed Round   
1052                             Verlinvest  8.000000e+06    Series A   
1053             SIG Global India Fund LLP.  8.043000e+06    Series C   
1054                      Chiratae Ventures  9.000000e+06    Series A   

     Unnamed: 9 Year of Funding  
0           NaN            2020  
1           NaN            2020  
2           NaN            2020  
3           NaN            2020  
4           NaN            2020  
...         ...             ...  
1050        NaN            2020  
1051        NaN            2020  
1052        NaN            2020  
1053        NaN            2020  
1054        NaN            2020  

[1055 rows x 11 columns]

In [83]:
# DROPPING UNNAMED COLUMN

startup_2020 = startup_2020.drop(columns = "Unnamed: 9")

In [84]:
startup_2020

Company/Brand Founded HeadQuarter              Sector  \
0       Aqgromalin    2019     Chennai            AgriTech   
1         Krayonnz    2019   Bangalore              EdTech   
2     PadCare Labs    2018        Pune  Hygiene management   
3            NCOME    2020   New Delhi              Escrow   
4       Gramophone    2016      Indore            AgriTech   
...            ...     ...         ...                 ...   
1050  Leverage Edu    2015       Delhi              Edtech   
1051         EpiFi    2015         NaN             Fintech   
1052       Purplle    2012      Mumbai           Cosmetics   
1053        Shuttl    2015       Delhi           Transport   
1054         Pando    2017     Chennai            Logitech   

                                           What it does  \
0                          Cultivating Ideas for Profit   
1     An academy-guardian-scholar centric ecosystem ...   
2      Converting bio-hazardous waste to harmless waste   
3                          Escrow-as-a-service platform   
4     Gramophone is an AgTech platform enabling acce...   
...                                                 ...   
1050  AI enabled marketplace that provides career gu...   
1051  It offers customers with a single interface fo...   
1052         Online makeup and beauty products retailer   
1053                    App based bus aggregator serice   
1054            Networked logistics management software   

                                               Founders  \
0                       Prasanna Manogaran, Bharani C L   
1                      Saurabh Dixit, Gurudutt Upadhyay   
2                                       Ajinkya Dhariya   
3                                         Ritesh Tiwari   
4     Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   
...                                                 ...   
1050                                  Akshay Chaturvedi   
1051                    Sujith Narayanan, Sumit Gwalani   
1052                          Manish Taneja, Rahul Dash   
1053                      Amit Singh, Deepanshu Malviya   
1054                     Jayakrishnan, Abhijeet Manohar   

                                   Investor     Amount($)       Stage  \
0                           Angel investors  2.000000e+05    Series A   
1                           GSF Accelerator  1.000000e+05    Pre-seed   
2                            Venture Center  8.586025e+07    Pre-seed   
3       Venture Catalysts, PointOne Capital  4.000000e+05    Series A   
4       Siana Capital Management, Info Edge  3.400000e+05    Series A   
...                                     ...           ...         ...   
1050  DSG Consumer Partners, Blume Ventures  1.500000e+06    Series A   
1051          Sequoia India, Ribbit Capital  1.320000e+07  Seed Round   
1052                             Verlinvest  8.000000e+06    Series A   
1053             SIG Global India Fund LLP.  8.043000e+06    Series C   
1054                      Chiratae Ventures  9.000000e+06    Series A   

     Year of Funding  
0               2020  
1               2020  
2               2020  
3               2020  
4               2020  
...              ...  
1050            2020  
1051            2020  
1052            2020  
1053            2020  
1054            2020  

[1055 rows x 10 columns]

In [85]:
file_name = "cleaned_2020.csv"

startup_2020.to_csv(file_name)

### STARTUP_2021 PROCESSING

In [86]:
startup_2021

Company/Brand  Founded HeadQuarter                 Sector  \
0     Unbox Robotics   2019.0   Bangalore             AI startup   
1             upGrad   2015.0      Mumbai                 EdTech   
2        Lead School   2012.0      Mumbai                 EdTech   
3            Bizongo   2015.0      Mumbai         B2B E-commerce   
4           FypMoney   2021.0    Gurugram                FinTech   
...              ...      ...         ...                    ...   
1204        Gigforce   2019.0    Gurugram  Staffing & Recruiting   
1205          Vahdam   2015.0   New Delhi       Food & Beverages   
1206    Leap Finance   2019.0   Bangalore     Financial Services   
1207    CollegeDekho   2015.0    Gurugram                 EdTech   
1208          WeRize   2019.0   Bangalore     Financial Services   

                                           What it does  \
0     Unbox Robotics builds on-demand AI-driven ware...   
1        UpGrad is an online higher education platform.   
2     LEAD School offers technology based school tra...   
3     Bizongo is a business-to-business online marke...   
4     FypMoney is Digital NEO Bank for Teenagers, em...   
...                                                 ...   
1204                  A gig/on-demand staffing company.   
1205  VAHDAM is among the world’s first vertically i...   
1206  International education loans for high potenti...   
1207  Collegedekho.com is Student’s Partner, Friend ...   
1208  India’s first socially distributed full stack ...   

                                               Founders  \
0                           Pramod Ghadge, Shahid Memon   
1     Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                            Smita Deorah, Sumeet Mehta   
3               Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                         Kapil Banwari   
...                                                 ...   
1204                        Chirag Mittal, Anirudh Syal   
1205                                         Bala Sarda   
1206                         Arnav Kumar, Vaibhav Singh   
1207                                       Ruchir Arora   
1208                      Vishal Chopra, Himanshu Gupta   

                                           Investor     Amount($)  \
0                       BEENEXT, Entrepreneur First    $1,200,000   
1          Unilazer Ventures, IIFL Asset Management  $120,000,000   
2                  GSV Ventures, Westbridge Capital   $30,000,000   
3                            CDC Group, IDG Capital   $51,000,000   
4     Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    $2,000,000   
...                                             ...           ...   
1204                                Endiya Partners      $3000000   
1205                                       IIFL AMC     $20000000   
1206                                   Owl Ventures     $55000000   
1207               Winter Capital, ETS, Man Capital     $26000000   
1208                 3one4 Capital, Kalaari Capital      $8000000   

             Stage  
0     Pre-series A  
1              NaN  
2         Series D  
3         Series C  
4             Seed  
...            ...  
1204  Pre-series A  
1205      Series D  
1206      Series C  
1207      Series B  
1208      Series A  

[1209 rows x 9 columns]

In [87]:
startup_2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1209 entries, 0 to 1208
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company/Brand  1209 non-null   object 
 1   Founded        1208 non-null   float64
 2   HeadQuarter    1208 non-null   object 
 3   Sector         1209 non-null   object 
 4   What it does   1209 non-null   object 
 5   Founders       1205 non-null   object 
 6   Investor       1147 non-null   object 
 7   Amount($)      1206 non-null   object 
 8   Stage          781 non-null    object 
dtypes: float64(1), object(8)
memory usage: 85.1+ KB


In [88]:
startup_2021.head()

Company/Brand  Founded HeadQuarter          Sector  \
0  Unbox Robotics   2019.0   Bangalore      AI startup   
1          upGrad   2015.0      Mumbai          EdTech   
2     Lead School   2012.0      Mumbai          EdTech   
3         Bizongo   2015.0      Mumbai  B2B E-commerce   
4        FypMoney   2021.0    Gurugram         FinTech   

                                        What it does  \
0  Unbox Robotics builds on-demand AI-driven ware...   
1     UpGrad is an online higher education platform.   
2  LEAD School offers technology based school tra...   
3  Bizongo is a business-to-business online marke...   
4  FypMoney is Digital NEO Bank for Teenagers, em...   

                                            Founders  \
0                        Pramod Ghadge, Shahid Memon   
1  Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                         Smita Deorah, Sumeet Mehta   
3            Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                      Kapil Banwari   

                                        Investor     Amount($)         Stage  
0                    BEENEXT, Entrepreneur First    $1,200,000  Pre-series A  
1       Unilazer Ventures, IIFL Asset Management  $120,000,000           NaN  
2               GSV Ventures, Westbridge Capital   $30,000,000      Series D  
3                         CDC Group, IDG Capital   $51,000,000      Series C  
4  Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    $2,000,000          Seed

In [89]:
# CHECKING FOR MISSING VALUES

startup_2021.isna().sum()

Company/Brand      0
Founded            1
HeadQuarter        1
Sector             0
What it does       0
Founders           4
Investor          62
Amount($)          3
Stage            428
dtype: int64

In [90]:
startup_2021.duplicated().sum()

19

### LET'S START FROM THE FOUNDED COLUMN FOR THE CLEANING

In [91]:
 startup_2021.Founded.unique() 



array([2019., 2015., 2012., 2021., 2014., 2018., 2016., 2020., 2010.,
       2017., 1993., 2008., 2013., 1999., 1989., 2011.,   nan, 2009.,
       2002., 1994., 2006., 2000., 2007., 1978., 2003., 1998., 1991.,
       1984., 2004., 2005., 1963.])

In [92]:
startup_2021.Founded.isna().sum()

1

### the Founded Column has only 1 nan value. We can use the mean or the mode to fill it 

In [93]:
# SETTING THE DATATYPE FOR THE FOUNDED COLUMN TO STRING

startup_2021['Founded'] = startup_2021['Founded'].replace(np.nan, 0)

startup_2021['Founded'] = startup_2021['Founded'].astype(int)

startup_2021['Founded']

0       2019
1       2015
2       2012
3       2015
4       2021
        ... 
1204    2019
1205    2015
1206    2019
1207    2015
1208    2019
Name: Founded, Length: 1209, dtype: int32

In [94]:
# calculating the mean and mode

startup_2021['Founded'].mean()

2014.987593052109

In [95]:
# mode

startup_2021['Founded'].mode()

0    2020
Name: Founded, dtype: int32

In [96]:
# Using the Mean to replace "0"

startup_2021.Founded = startup_2021.Founded.replace("0",2015)

startup_2021.Founded

0       2019
1       2015
2       2012
3       2015
4       2021
        ... 
1204    2019
1205    2015
1206    2019
1207    2015
1208    2019
Name: Founded, Length: 1209, dtype: int32

In [97]:
startup_2021.Founded.isna().sum()

0

### Our Founded Column is clean now.

### CLEANING OF THE HEADQUARTERS COLUMN

In [98]:
startup_2021['HeadQuarter'].unique()

array(['Bangalore', 'Mumbai', 'Gurugram', 'New Delhi', 'Hyderabad',
       'Jaipur', 'Ahmadabad', 'Chennai', nan,
       'Small Towns, Andhra Pradesh', 'Goa', 'Rajsamand', 'Ranchi',
       'Faridabad, Haryana', 'Gujarat', 'Pune', 'Thane', 'Computer Games',
       'Cochin', 'Noida', 'Chandigarh', 'Gurgaon', 'Vadodara',
       'Food & Beverages', 'Pharmaceuticals\t#REF!', 'Gurugram\t#REF!',
       'Kolkata', 'Ahmedabad', 'Mohali', 'Haryana', 'Indore', 'Powai',
       'Ghaziabad', 'Nagpur', 'West Bengal', 'Patna', 'Samsitpur',
       'Lucknow', 'Telangana', 'Silvassa', 'Thiruvananthapuram',
       'Faridabad', 'Roorkee', 'Ambernath', 'Panchkula', 'Surat',
       'Coimbatore', 'Andheri', 'Mangalore', 'Telugana', 'Bhubaneswar',
       'Kottayam', 'Beijing', 'Panaji', 'Satara', 'Orissia', 'Jodhpur',
       'New York', 'Santra', 'Mountain View, CA', 'Trivandrum',
       'Jharkhand', 'Kanpur', 'Bhilwara', 'Guwahati',
       'Online Media\t#REF!', 'Kochi', 'London',
       'Information Technolo

In [99]:
startup_2021['HeadQuarter'].isna().sum()

1

In [100]:
startup_2021['HeadQuarter'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 1209 entries, 0 to 1208
Series name: HeadQuarter
Non-Null Count  Dtype 
--------------  ----- 
1208 non-null   object
dtypes: object(1)
memory usage: 9.6+ KB


In [101]:
# Replacing NAN values with "Undefined"

headquarters_2021 = startup_2021.HeadQuarter.isnull()

headquarters_2021


0       False
1       False
2       False
3       False
4       False
        ...  
1204    False
1205    False
1206    False
1207    False
1208    False
Name: HeadQuarter, Length: 1209, dtype: bool

In [102]:
# Replacing the 'NAN' value with Keyword "Undefined"

for i in range(len(headquarters_2021)):
    if headquarters_2021[i] == True:
        startup_2021.HeadQuarter[i] = "Undefined"

       

C:\Users\User\AppData\Local\Temp\ipykernel_23912\1738805364.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  startup_2021.HeadQuarter[i] = "Undefined"


In [103]:
startup_2021.HeadQuarter.isnull().sum()

0

### Now, the Headquarters Column is great.


### LET'S LOOK AT THE FOUNDERS COLUMN TO CLEAN IT

In [104]:
startup_2021['Founders'].isna().sum()

4

In [105]:
startup_2021['Founders'].unique()

array(['Pramod Ghadge, Shahid Memon',
       'Mayank Kumar, Phalgun Kompalli, Ravijot Chugh, Ronnie Screwvala',
       'Smita Deorah, Sumeet Mehta', ..., 'Bala Sarda',
       'Arnav Kumar, Vaibhav Singh', 'Vishal Chopra, Himanshu Gupta'],
      dtype=object)

In [106]:
founders_2021 = startup_2021.Founders.isnull()

founders_2021

0       False
1       False
2       False
3       False
4       False
        ...  
1204    False
1205    False
1206    False
1207    False
1208    False
Name: Founders, Length: 1209, dtype: bool

In [107]:
# Replacing the '...' value with Keyword "Undefined"

for i in range(len(founders_2021)):
    if founders_2021[i] == True:
        startup_2021.Founders[i] = "Undefined"

C:\Users\User\AppData\Local\Temp\ipykernel_23912\2680666504.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  startup_2021.Founders[i] = "Undefined"


In [108]:
startup_2021.Founders.isnull().sum()

0

### The Founder column is Cleaned now


### CLEANING OF THE INVESTOR COLUMN

In [109]:
startup_2021.Investor.info()

<class 'pandas.core.series.Series'>
RangeIndex: 1209 entries, 0 to 1208
Series name: Investor
Non-Null Count  Dtype 
--------------  ----- 
1147 non-null   object
dtypes: object(1)
memory usage: 9.6+ KB


In [110]:

investors_2021 = startup_2021.Investor.isna().sum()

investors_2021

62

In [126]:
startup_2021['Investor'] = startup_2021['Investor'].apply(str)


# Replacing the 'missing' values with Keyword "Unknown"
#for i in range(investors_2021):
    #if investors_2021[i] == True:
       # startup_2021.Investor[i] = "unknown"
        
#startup_2021.Investor.info()

In [127]:
startup_2021.isna().sum()

Company/Brand      0
Founded            0
HeadQuarter        0
Sector             0
What it does       0
Founders           0
Investor           0
Amount($)         51
Stage            428
dtype: int64

### The Investor column is now Clean.


# CLEANING OF THE AMOUNT COLUMN

In [128]:
startup_2021['Amount($)'].isna().sum()

51

In [129]:
startup_2021['Amount($)'].unique()

array(['$1200000', '$120000000', '$30000000', '$51000000', '$2000000',
       '$188000000', '$200000', nan, '$1000000', '$3000000', '$100000',
       '$700000', '$9000000', '$40000000', '$49000000', '$400000',
       '$300000', '$25000000', '$160000000', '$150000', '$1800000',
       '$5000000', '$850000', '$53000000', '$500000', '$1100000',
       '$6000000', '$800000', '$10000000', '$21000000', '$7500000',
       '$26000000', '$7400000', '$1500000', '$600000', '$800000000',
       '$17000000', '$3500000', '$15000000', '$215000000', '$2500000',
       '$350000000', '$5500000', '$83000000', '$110000000', '$500000000',
       '$65000000', '$150000000000', '$300000000', '$2200000',
       '$35000000', '$140000000', '$4000000', '$13000000', 'nan',
       '$Undisclosed', '$9500000', '$8000000', '$12000000', '$1700000',
       '$150000000', '$100000000', '$225000000', '$6700000', '$1300000',
       '$20000000', '$250000', '$52000000', '$3800000', '$17500000',
       '$42000000', '$2300000',

In [115]:
# converting the datatype to string

startup_2021['Amount($)'] = startup_2021['Amount($)'].apply(str)


# Removing the comas and spaces

startup_2021['Amount($)'].replace(',', '', inplace = True, regex = True)

startup_2021

Company/Brand  Founded HeadQuarter                 Sector  \
0     Unbox Robotics     2019   Bangalore             AI startup   
1             upGrad     2015      Mumbai                 EdTech   
2        Lead School     2012      Mumbai                 EdTech   
3            Bizongo     2015      Mumbai         B2B E-commerce   
4           FypMoney     2021    Gurugram                FinTech   
...              ...      ...         ...                    ...   
1204        Gigforce     2019    Gurugram  Staffing & Recruiting   
1205          Vahdam     2015   New Delhi       Food & Beverages   
1206    Leap Finance     2019   Bangalore     Financial Services   
1207    CollegeDekho     2015    Gurugram                 EdTech   
1208          WeRize     2019   Bangalore     Financial Services   

                                           What it does  \
0     Unbox Robotics builds on-demand AI-driven ware...   
1        UpGrad is an online higher education platform.   
2     LEAD School offers technology based school tra...   
3     Bizongo is a business-to-business online marke...   
4     FypMoney is Digital NEO Bank for Teenagers, em...   
...                                                 ...   
1204                  A gig/on-demand staffing company.   
1205  VAHDAM is among the world’s first vertically i...   
1206  International education loans for high potenti...   
1207  Collegedekho.com is Student’s Partner, Friend ...   
1208  India’s first socially distributed full stack ...   

                                               Founders  \
0                           Pramod Ghadge, Shahid Memon   
1     Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                            Smita Deorah, Sumeet Mehta   
3               Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                         Kapil Banwari   
...                                                 ...   
1204                        Chirag Mittal, Anirudh Syal   
1205                                         Bala Sarda   
1206                         Arnav Kumar, Vaibhav Singh   
1207                                       Ruchir Arora   
1208                      Vishal Chopra, Himanshu Gupta   

                                           Investor   Amount($)         Stage  
0                       BEENEXT, Entrepreneur First    $1200000  Pre-series A  
1          Unilazer Ventures, IIFL Asset Management  $120000000           NaN  
2                  GSV Ventures, Westbridge Capital   $30000000      Series D  
3                            CDC Group, IDG Capital   $51000000      Series C  
4     Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    $2000000          Seed  
...                                             ...         ...           ...  
1204                                Endiya Partners    $3000000  Pre-series A  
1205                                       IIFL AMC   $20000000      Series D  
1206                                   Owl Ventures   $55000000      Series C  
1207               Winter Capital, ETS, Man Capital   $26000000      Series B  
1208                 3one4 Capital, Kalaari Capital    $8000000      Series A  

[1209 rows x 9 columns]

In [124]:
# Replacing Undisclosed, Series C, Seed, ah! Ventures, Pre-series A, ITO Angel Network, LetsVenture with nan

startup_2021['Amount($)'] = startup_2021['Amount($)'].replace('Undisclosed',np.nan)

startup_2021['Amount($)'] = startup_2021['Amount($)'].replace('Series C',np.nan)
startup_2021['Amount($)'] = startup_2021['Amount($)'].replace('Seed',np.nan)
startup_2021['Amount($)'] = startup_2021['Amount($)'].replace('ah! Ventures',np.nan)
startup_2021['Amount($)'] = startup_2021['Amount($)'].replace('Pre-series A',np.nan)
startup_2021['Amount($)'] = startup_2021['Amount($)'].replace('ITO Angel Network',np.nan)
startup_2021['Amount($)'] = startup_2021['Amount($)'].replace('LetsVenture',np.nan)
startup_2021['Amount($)'] = startup_2021['Amount($)'].replace('Upsparks',np.nan)

In [125]:
startup_2021['Amount($)'].unique()

array(['$1200000', '$120000000', '$30000000', '$51000000', '$2000000',
       '$188000000', '$200000', nan, '$1000000', '$3000000', '$100000',
       '$700000', '$9000000', '$40000000', '$49000000', '$400000',
       '$300000', '$25000000', '$160000000', '$150000', '$1800000',
       '$5000000', '$850000', '$53000000', '$500000', '$1100000',
       '$6000000', '$800000', '$10000000', '$21000000', '$7500000',
       '$26000000', '$7400000', '$1500000', '$600000', '$800000000',
       '$17000000', '$3500000', '$15000000', '$215000000', '$2500000',
       '$350000000', '$5500000', '$83000000', '$110000000', '$500000000',
       '$65000000', '$150000000000', '$300000000', '$2200000',
       '$35000000', '$140000000', '$4000000', '$13000000', 'nan',
       '$Undisclosed', '$9500000', '$8000000', '$12000000', '$1700000',
       '$150000000', '$100000000', '$225000000', '$6700000', '$1300000',
       '$20000000', '$250000', '$52000000', '$3800000', '$17500000',
       '$42000000', '$2300000',